celery/app/amqp.py

```diff
@@ -213,7 +213,7 @@ def publish_task(self, task_name, task_args=None, task_kwargs=None,
         retry = self.retry if retry is None else retry
         _rp = (dict(self.retry_policy, **retry_policy) if retry_policy
                else self.retry_policy)
-        task_id = task_id or uuid()
+        task_id = task_id or self.app.tasks[task_name].generate_task_id() # is this valid?
```

Comment:
No, you can't do that - because at this point you don't have any task *object*, only the string representation. And you may send a string representation of a task to a remote worker that has that specific task implementation, while the caller side may has no such task implementation at all.